code/game/objects/items/granters.dm

```diff
@@ -341,11 +341,9 @@
 	if(!martial)
 		return
 	var/datum/martial_art/MA = new martial
-	if(user.mind.martial_art)
-		for(var/datum/martial_art/knownmartial in user.mind.martial_art)
-			if(knownmartial.type == MA.type)
-				to_chat(user,"<span class='warning'>You already know [martialname]!</span>")
-				return
+	if(user.mind.has_martialart(MA.name))
+		to_chat(user,"<span class='warning'>You already know [martialname]!</span>")
```

Comment:
```suggestion
 		to_chat(user,"<span class='warning'>You already know [MA.name]!</span>")
```

Reply:
the `martialname` var can be removed entirely too i think

Reply:
Looking at it, some of the martialname stuff is for fluff, but necessary fluff. Like the CQC book has martial name as close quarters combat